utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -477,21 +485,26 @@ impl State {
 				Builder::<Block>::new().mode(Mode::Offline(OfflineConfig {
 					state_snapshot: SnapshotConfig::new(snapshot_path),
 				})),
-			State::Live { snapshot_path, pallets, uri, at } => {
+			State::Live { snapshot_path, pallets, uri, at, child_tree } => {
 				let at = match at {
 					Some(at_str) => Some(hash_of::<Block>(at_str)?),
 					None => None,
 				};
-				Builder::<Block>::new()
+				let mut builder = Builder::<Block>::new()
 					.mode(Mode::Online(OnlineConfig {
 						transport: uri.to_owned().into(),
 						state_snapshot: snapshot_path.as_ref().map(SnapshotConfig::new),
-						pallets: pallets.to_owned().unwrap_or_default(),
+						pallets: pallets.clone().unwrap_or_default(),
 						at,
+						..Default::default()
 					}))
 					.inject_hashed_key(
 						&[twox_128(b"System"), twox_128(b"LastRuntimeUpgrade")].concat(),
-					)
+					);
+				if child_tree.unwrap_or(false) || pallets.is_none() {
+					builder = builder.inject_default_child_tree_prefix();
```

Comment:
Not sure, but when pallets.is_none, we fetch the whole top state (empty prefix), so need to add specifically child tree prefix ?

Reply:
yeah probably not needed, will try and remove if not needed.